### ISSUE TITLE
country on Address subrecord requires the abbreviation string rather than the IID of the Country record

### ISSUE THREAD

**Issue by steveklett** [ExploreConsulting/netsuite-fasttrack-toolkit-ss2#49]
When setting the country property on an Address subrecord I'm getting error:

> You have entered an Invalid Field Value 227 for the following field: country

I've verified that it's the abbreviation value (e.g., 'US') that is required. The NetSuite record browser states this field is a select, and it is, but why it won't take an IID I cannot say.
The workaround is to hit the underlying ns record instance like so:

```
newRec.nsrecord.getSubrecord({fieldId: 'shippingaddress'}).setValue({
    fieldId: `country`,
    value: getCountryById(230).abbrev
})
```

**Comment by ShawnTalbert** [comment]
Yes, another NetSuite-ism. The easiest solution here it to make that a string or perhaps `string | number` in case NS silently changes how that select works...

Another temporary workaround would be

```typescript
newRec.shippingaddress.country = getCountryById(230)!.abbrev as any
```

**Comment by steveklett** [comment]
I plan to make the change and submit a PR, operative word is 'plan'
I didn't realize you could cast to any like that, much clean, thanks for the tip.
Hope all is well with you. :)

**Comment by ShawnTalbert** [comment]
Aye, use `any` as a last resort, but once in a while you have no other
great choice. You could probably also cast to `unknown` then to `number`
but that's even more confusing.

**Comment by ShawnTalbert** [comment]
should be resolved in 6.0.0, reopen if needed
